### PR TITLE
Feat: Add thumbnail previews to all files

### PR DESCRIPTION
**Pull request by iswiftshek**
- Used SDWebImage to fetch and cache the images. 
- Sample thumbnails added to all types of files (image, audio, video, others).

Issue: #145 Solved.

### CODE REVIEW

File: AmahiAnywhere/AmahiAnywhere/Presentation/Files/FilesViewController+UITableViewDelegates.swift [hunk 1]
```diff
@@ -26,6 +27,21 @@ extension FilesViewController : UITableViewDelegate, UITableViewDataSource {
...
+            else if (serverFile.mime_type?.starts(with: "audio"))! {
+                imageName = "audio"
+            }
+            else if (serverFile.mime_type?.starts(with: "video"))! {
+                imageName = "video"
+            }
+            else {
+                imageName = "file"
+            }
+            cell.thumbnailImage.sd_setImage(with: URL(string: ServerApi.shared!.getFileUri(serverFile).absoluteString), placeholderImage: UIImage(named: imageName))
```
Comment (csoni111) [inline AmahiAnywhere/AmahiAnywhere/Presentation/Files/FilesViewController+UITableViewDelegates.swift]: Should use the `options:SDWebImageRefreshCached` to respect caching headers sent by server. Read the `Handle image refresh` section in the docs [here](http://cocoadocs.org/docsets/SDWebImage/4.1.2/).
For swift check this [answer](https://stackoverflow.com/a/43814202/6352482).

**Comment by csoni111** [comment]
Currently we have both Imaginary (used by lightbox) and SDWebImage. They both cache images separately which is bad. We can use [this feature](https://github.com/hyperoslo/Lightbox#image-loading) of Lightbox to load the images using SDWebImage only inside lightbox too and remove Imaginary.

**Comment by iswiftshek** [comment]
> Currently we have both Imaginary (used by lightbox) and SDWebImage. They both cache images separately which is bad. We can use [this feature](https://github.com/hyperoslo/Lightbox#image-loading) of Lightbox to load the images using SDWebImage only inside lightbox too and remove Imaginary.

Yes, right. But doing so.. 

ezgif com-resize (2)

the indicator won't stop loading. I tried doing a few things, but I couldn't fix it. 
How do we resolve this? @csoni111

**Comment by csoni111** [comment]
@iswiftshek https://github.com/hyperoslo/Lightbox/issues/92#issuecomment-280798724

**Comment by iswiftshek** [comment]
Done @csoni111 @cpg 
SDWebImage is used for caching, and the loading indicator works well.

**Comment by codedentwickler** [comment]
Well done @iswiftshek , I'm testing this PR with the Demo server currently and only the place holder show. Is it loading the thumbnail on your end.

**Pull request merged**